### ISSUE TITLE
Seed / reproducibility

### ISSUE THREAD

**Issue by vince62s** [tensorflow/tensor2tensor#485]
even though this https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/bin/t2t-trainer#L173
was added in 1.4.0, two runs in ar ow of the same config does not produce the sames results.
Just checked loss and evals afetr 2000 steps, different.

**Comment by martinpopel** [comment]
Note that t2t-datagen uses more sophisticated approach than t2t-trainer: https://github.com/tensorflow/tensor2tensor/blob/758991dd35abc510e93caa5c01c6476bb2380b6e/tensor2tensor/bin/t2t-datagen#L121-L125
Perhaps the same approach (setting all three random seeds and keep the seed as a parameter, possibly with 0 meaning, no fixing) should be done in t2t-trainer and t2t-decoder.
@vince62s: can you try it and report if it helps?

**Comment by vince62s** [comment]
In the previous version, when the line of code was not there, I had tried these lines in t2t-trainer
did not help (as reported in gitter).

**Comment by rsepassi** [comment]
Adding the fuller seed setting to the trainer. Seems like a good idea, even if it isn't the whole solution.

**Comment by jiangbojian** [comment]
You can set add 'tf_random_seed=1' in Runconfig

**Comment by jiangbojian** [comment]
You can set add 'tf_random_seed=1' in Runconfig

  estimator = tf.contrib.learn.Estimator(
      model_fn=model_builder(model_name, hparams=hparams),
      model_dir=output_dir,
      config=tf.contrib.learn.RunConfig(
          master=FLAGS.master,
          model_dir=output_dir,
          gpu_memory_fraction=FLAGS.worker_gpu_memory_fraction,
          session_config=session_config(),
          keep_checkpoint_max=FLAGS.keep_checkpoint_max,tf_random_seed=1))

**Comment by jiangbojian** [comment]
You can set add 'tf_random_seed=1' in Runconfig

estimator = tf.contrib.learn.Estimator(
model_fn=model_builder(model_name, hparams=hparams),
model_dir=output_dir,
config=tf.contrib.learn.RunConfig(
master=FLAGS.master,
model_dir=output_dir,
gpu_memory_fraction=FLAGS.worker_gpu_memory_fraction,
session_config=session_config(),
keep_checkpoint_max=FLAGS.keep_checkpoint_max,tf_random_seed=1))

now, it works when initialize the tensor!
 i set all dropout=0.0 and close shuffle when training,but when i rerun this model, the loss and evals are different

**Comment by jiangbojian** [comment]
You can set add 'tf_random_seed=1' in Runconfig,the code as follows:

estimator = tf.contrib.learn.Estimator(
model_fn=model_builder(model_name, hparams=hparams),
model_dir=output_dir,
config=tf.contrib.learn.RunConfig(
master=FLAGS.master,
model_dir=output_dir,
gpu_memory_fraction=FLAGS.worker_gpu_memory_fraction,
session_config=session_config(),
keep_checkpoint_max=FLAGS.keep_checkpoint_max,tf_random_seed=1))



now, it works when initialize the tensor!
i set all dropout=0.0 and close shuffle when training,but when i rerun this model, the loss and evals are different

**Comment by jiangbojian** [comment]
@vince62s  the other solution,you can add 'set_random_seed()'  to  'def run()'  of  'trainer_util.py'

but,i don't know why it not works when I had tried these lines in 't2t-trainer'

**Comment by rsepassi** [comment]
Thanks @jiangbojian! Yes, I think the issue is that we're not setting it in the `RunConfig`. Will have that out in the next release.

**Comment by rsepassi** [comment]
This is now set in the RunConfig. Try your commands again and let me know how it goes.
